src/login_nopam.c

```diff
@@ -265,41 +263,25 @@ static bool user_match (const char *tok, const char *string)
 
 static const char *resolve_hostname (const char *string)
 {
-	int                  gai;
-	char                 *p;
-	struct addrinfo      *addrs;
-	struct sockaddr      *sa;
-	struct sockaddr_in   *sin;
-	struct sockaddr_in6  *sin6;
-
-	gai = getaddrinfo(string, NULL, NULL, &addrs);
-	if (gai != 0)
+	int              gai_err;
+	char             *addr_str;
+	struct addrinfo  *addrs;
+
+	gai_err = getaddrinfo(string, NULL, NULL, &addrs);
+	if (gai_err != 0)
 		goto notfound;
 
-	sa = addrs[0].ai_addr;
-	switch (addrs[0].ai_addr->sa_family) {
-	case AF_INET:
-		sin = (struct sockaddr_in *) sa;
-		inet_ntop(AF_INET, &sin->sin_addr,
-		          inet_ntop_buffer, NITEMS(inet_ntop_buffer));
-		p = inet_ntop_buffer;
-		break;
-	case AF_INET6:
-		sin6 = (struct sockaddr_in6 *) sa;
-		inet_ntop(AF_INET6, &sin6->sin6_addr,
-		          inet_ntop_buffer, NITEMS(inet_ntop_buffer));
-		p = inet_ntop_buffer;
-		break;
-	default:
+	addr_str = inet_sockaddr2str(addrs[0].ai_addr);
+	if (addr_str == NULL) {
 		SYSLOG ((LOG_ERR, "Hypothetical future IPv7?"));
```

Comment:
You don't need to print this here as it was already printed in `inet_sockaddr2str()`.

Reply:
No, I didn't log in inet_sockaddr2str();  I only set errno.

Reply:
Ahh, I see where you got confused.  The log in `inet_sockaddr2str()` is for when no addresses are found.  This one is for when an IPv7? address is found.